fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/AggStateFunctionBuilder.java

```diff
@@ -0,0 +1,126 @@
+                return false;
+            }
+
+            return nestedBuilder.canApply(((AggStateType) argument.getDataType()).getSubTypes().stream().map(t -> {
+                return new SlotReference("mocked", t);
+            }).collect(Collectors.toList()));
+        }
+    }
+
+    private BoundFunction buildState(String nestedName, List<? extends Object> arguments) {
+        return nestedBuilder.build(nestedName, arguments);
+    }
+
+    private BoundFunction buildMergeOrUnion(String nestedName, List<? extends Object> arguments) {
+        if (arguments.size() != 1 || !(arguments.get(0) instanceof Expression)
+                || !((Expression) arguments.get(0)).getDataType().isAggStateType()) {
+            String argString = arguments.stream().map(arg -> {
+                if (arg == null) {
+                    return "null";
+                } else if (arg instanceof Expression) {
+                    return ((Expression) arg).toSql();
+                } else {
+                    return arg.toString();
+                }
+            }).collect(Collectors.joining(", ", "(", ")"));
+            throw new IllegalStateException("Can not build AggState nested function: '" + nestedName + "', expression: "
+                    + nestedName + argString);
+        }
+
+        Expression arg = (Expression) arguments.get(0);
+        AggStateType type = (AggStateType) arg.getDataType();
+
+        List<Expression> nestedArgumens = type.getSubTypes().stream().map(t -> {
+            return new SlotReference("mocked", t);
+        }).collect(Collectors.toList());
+
+        return nestedBuilder.build(nestedName, nestedArgumens);
+    }
+
+    @Override
+    public BoundFunction build(String name, List<? extends Object> arguments) {
+        String nestedName = getNestedName(name);
+        if (combinatorSuffix.equals(STATE)) {
+            BoundFunction nestedFunction = buildState(nestedName, arguments);
+            return new StateCombinator((List<Expression>) arguments, nestedFunction);
+        } else {
+            BoundFunction nestedFunction = buildMergeOrUnion(nestedName, arguments);
+            return new MergeCombinator((List<Expression>) arguments, nestedFunction);
+        }
+    }
+
+    @Override
+    public String toString() {
+        return combinatorSuffix + "(" + nestedBuilder.toString() + ")";
+    }
+
+    public static boolean isAggStateCombinator(String name) {
+        return name.toLowerCase().endsWith(COMBINATOR_LINKER + STATE)
+                || name.toLowerCase().endsWith(COMBINATOR_LINKER + MERGE)
+                || name.toLowerCase().endsWith(COMBINATOR_LINKER + UNION);
+    }
+
+    public static String getNestedName(String name) {
+        return name.substring(0, name.lastIndexOf(COMBINATOR_LINKER));
+    }
+
+    public static String getCombinatorSuffix(String name) {
+        return name.substring(name.lastIndexOf(AggStateFunctionBuilder.COMBINATOR_LINKER) + 1);
```

Comment:
should check name contains COMBINATOR_LINKER first, and throw exception if not contains it